kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DomainMetadataSuite.scala

```diff
@@ -0,0 +1,443 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.defaults
+
+import io.delta.kernel._
+import io.delta.kernel.data.Row
+import io.delta.kernel.defaults.internal.parquet.ParquetSuiteBase
+import io.delta.kernel.engine.Engine
+import io.delta.kernel.exceptions._
+import io.delta.kernel.internal.SnapshotImpl
+import io.delta.kernel.internal.actions.{DomainMetadata, Protocol, SingleAction}
+import io.delta.kernel.internal.util.Utils.toCloseableIterator
+import io.delta.kernel.utils.CloseableIterable
+import io.delta.kernel.utils.CloseableIterable.{emptyIterable, inMemoryIterable}
+
+import java.util.Collections
+import scala.collection.JavaConverters._
+import scala.collection.immutable.Seq
+
+class DomainMetadataSuite extends DeltaTableWriteSuiteBase with ParquetSuiteBase {
```

Comment:
Good tests!

We'll want to also have an integration test with spark:
- Create a table with Delta spark containing DomainMetadata actions, read it with Delta kernel
One way you can do this is by generating a "golden table" using Delta spark: https://github.com/delta-io/delta/blob/1eff5df1e70a3e78a2801174a7e7b90cd1c91bfd/connectors/golden-tables/src/test/scala/io/delta/golden/GoldenTables.scala#L61

then read it using kernel, see for example https://github.com/delta-io/delta/blob/1eff5df1e70a3e78a2801174a7e7b90cd1c91bfd/kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/LogReplaySuite.scala#L41

Reply:
Added an integration test.